### DISCUSSION TITLE
Allow specification of graphql conversion functions for scalars.

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#223] [Ideas]
Inside the `scalars` settings, we should be able to define a conversion function for the scalar to a graphql object. This is similar to how the "model to database" conversion methods work (ie. `modelToDB` and `dbToModel`)

**EXAMPLE**

```typescript
...
scalars: {
  Json: {
    modelToGql: (value: unknown) => JSON.stringify();
    gqlToModel: (value: unknown) => JSON.parse(unknown);
  }
}
```

We can implement this behavior at the moment by directly using [Apollo's API](https://www.apollographql.com/docs/apollo-server/schema/custom-scalars/#example-the-date-scalar).

```typescript
const { GraphQLScalarType, Kind } = require('graphql');

const dateScalar = new GraphQLScalarType({
  name: 'Date',
  description: 'Date custom scalar type',
  serialize(value) {
    return value.getTime(); // Convert outgoing Date to integer for JSON
  },
  parseValue(value) {
    return new Date(value); // Convert incoming integer to Date
  },
  parseLiteral(ast) {
    if (ast.kind === Kind.INT) {
      return new Date(parseInt(ast.value, 10)); // Convert hard-coded AST string to integer and then to Date
    }
    return null; // Invalid hard-coded value (not an integer)
  },
});
```

**Comment by edobrb** [comment]
It is not exactly clear to me in which scenario this feature can come in handy. Whereas Typetta can also be used without necessarily having a GraphQL backend.

**Reply by Atlinx** [reply]
Yeah that makes sense, this might be too specialized to be put in a general ORM.
